lib/enumerable_2.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_count(item = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield(i) }
+    elsif !item.nil?
+      my_each { |i| count += 1 if i == item }
+    else
+      return size
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    arr = []
+    if proc
+      my_each { |i| arr.push(proc.call(i)) }
+    elsif block_given?
+      my_each { |i| arr.push(yield(i)) }
+    else
+      return my_each
+    end
+    arr
+  end
+
+  def my_inject(*args)
+    values = get_initial_symbol(args)
+    initial = values[0]
+    sym = values[1]
+
+    if block_given?
+      my_each_with_index do |item, index|
+        initial = args.empty? && index.zero? ? initial : yield(initial, item)
+      end
+    else
+      my_each_with_index do |item, index|
+        initial = args.size == 1 && index.zero? ? initial : initial.send(sym, item)
+      end
+    end
+    initial
+  end
```

Comment:
**[REQUIRED]**
* Combines all elements of enum by applying a binary operation, specified by a block.
![Screenshot from 2020-01-06 01-40-02](https://user-images.githubusercontent.com/44072711/71800275-a5b13a00-3025-11ea-8cb8-e265e34af0d4.png)

* When a symbol is specified combines each element of the collection by applying the symbol as a named method.
![Screenshot from 2020-01-06 01-42-45](https://user-images.githubusercontent.com/44072711/71800362-ec9f2f80-3025-11ea-921c-9404cea55168.png)